### PR TITLE
Implement redux provider in App.js

### LINKED ISSUE #236: Implement redux Provider in App.js
For test purposes a redux Provider has to be implemented in `App.js` to be able to use redux containers.

A best practice is to implement a `configureStore.js` in which the store is created and enhancers like middlewares are applied. The `redux-thunk` middleware should be applied.

Since we use react-hot-loader `configureStore.js` should handle hot reloading in the following way:
```
import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";

import rootReducer from "../reducers";

// Middleware you want to use in development:
const middleware = applyMiddleware(thunk);
const enhancer = compose(
    middleware,
);

export default function configureStore(history, initialState) {
    
    const store = createStore(rootReducer, initialState, enhancer);

    if (module.hot) {
        // Enable Webpack hot module replacement for reducers
        module.hot.accept("../reducers", () => {
            const nextRootReducer = require("../reducers");
            store.replaceReducer(nextRootReducer);
        });
    }

    return store;
}
```

### PR DESCRIPTION
**Pull request by Xyfi**
Adds a global redux provider to `App.js` for the test environment.

Hot reloading has been implemented for reloading reducers in `configureStore.js`.

`Redux devtools` have been implemented for easily debugging state changes. Press **Ctrl + H** to open the devtools.

## Test instructions
Replace the contents of `ContentAnalysis.js` with the following content:

Open the browser and press **Ctrl + H** to see the initial state. Now press the button on the `Content analysis` tab. You should see the state change in the devtools and a list on screen being updated.
```js
import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { connect } from "react-redux";
import { updateSeoResult } from "../actions/contentAnalysis";

export const ContentAnalysisContainer = styled.div`
	min-height: 700px;
	width: 100%;
	background-color: white;
`;

/**
 * Returns the ContentAnalysis component.
 *
 * @returns {ReactElement} The ContentAnalysis component.
 */
class ContentAnalysis extends React.Component {
	updateSeoResult() {
		const keyword = "keyword";
		let length = 0;
		if( this.props.seo[ keyword ] ) {
			length = this.props.seo[ keyword ].length;
		}
		this.props.updateSeoResult( keyword, {
			id: `resultId${ length }`,
		} );
	}

	renderKeywordResults() {
		const keyword = "keyword";
		if( ! this.props.seo[ keyword ] ) {
			return null;
		}
		const results = this.props.seo[ keyword ];
		return results.map( ( result ) => {
			return (
				<li key={ result.id }>{ result.id }</li>
			);
		} );
	}

	render() {
		return (
			<ContentAnalysisContainer>
				<button onClick={ this.updateSeoResult.bind( this ) }>Add keyword result</button>
				<ol>
					{ this.renderKeywordResults() }
				</ol>
			</ContentAnalysisContainer>
		);
	};
}

ContentAnalysis.propTypes = {
	updateSeoResult: PropTypes.func,
	seo: PropTypes.object,
};

export default connect(
	state => ( {
		seo: state.contentAnalysis.seo,
	} ),
	{
		updateSeoResult,
	}
)( ContentAnalysis );
```

Fixes #236

### CODE REVIEW

File: app/TopLevelComponents.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Provider } from "react-redux";
+
+import DevTools from "./utils/DevTools";
+import configureStore from "./configureStore";
+
+const store = configureStore();
+
+const TopLevelComponents = ( { children } ) => (
```
Comment (IreneStr) [inline app/TopLevelComponents.js]: Could you make this name more specific?

File: app/utils/DevTools.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+// Exported from redux-devtools
+import { createDevTools } from "redux-devtools";
+
+// Monitors are separate packages, and you can make a custom one
+import LogMonitor from "redux-devtools-log-monitor";
+import DockMonitor from "redux-devtools-dock-monitor";
+
+// createDevTools takes a monitor and produces a DevTools component
+const DevTools = createDevTools(
+	// Monitors are individually adjustable with props.
```
Comment (IreneStr) [inline app/utils/DevTools.js]: Please use /* ... */ for multiline comments

File: app/utils/DevTools.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+	// Consult their repositories to learn about those props.
+	// Here, we put LogMonitor inside a DockMonitor.
+	// Note: DockMonitor is visible by default.
+	<DockMonitor toggleVisibilityKey="ctrl-h"
+				 changePositionKey="ctrl-q"
+				 defaultIsVisible={false}>
```
Comment (IreneStr) [inline app/utils/DevTools.js]: Spaces

File: render.js [hunk 4]
```diff
@@ -2,23 +2,27 @@ import "babel-polyfill";
...
+		document.getElementById( "container" )
 	);
 }
 
 render( App );
 
 if ( module.hot ) {
 	module.hot.accept( "./App", () => {
-		const NextRoot = require( "./App" ).default;
+		const NextRoot = require( "./App" ).default; // eslint-disable-line global-require
```
Comment (IreneStr) [inline render.js]: Move comment to above the code line

File: app/utils/DevTools.js (hunk 3 above)
Comment (IreneStr) [inline app/utils/DevTools.js]: And please check indentation (might be GitHub's fault)

File: app/utils/DevTools.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
+import React from "react";
+
+// Exported from redux-devtools
+import { createDevTools } from "redux-devtools";
+
+// Monitors are separate packages, and you can make a custom one
+import LogMonitor from "redux-devtools-log-monitor";
+import DockMonitor from "redux-devtools-dock-monitor";
+
+// createDevTools takes a monitor and produces a DevTools component
```
Comment (IreneStr) [inline app/utils/DevTools.js]: Please start comments in uppercase to avoid (future) eslint errors.

File: app/TopLevelComponents.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+	<Provider store={ store }>
+		<div>
+			{ children }
+			<DevTools />
+		</div>
+	</Provider>
+);
+
+TopLevelComponents.propTypes = {
+	children: PropTypes.node,
```
Comment (IreneStr) [inline app/TopLevelComponents.js]: Please set default or make required

File: app/TopLevelComponents.js (hunk 1 above)
Comment (IreneStr) [inline app/TopLevelComponents.js]: AppWrapper?

**Comment by IreneStr** [comment]
CR done 👍

**Comment by boblinthorst** [comment]
Acceptance: works as described.

**Pull request merged**